cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -583,3 +583,76 @@ def test_guest_os_shutdown(appliance, testing_vm_tools, ensure_vm_running, soft_
+    """This fixture is used to create archived or orphaned VM"""
+    if request.param == "archived_vm":
+        # Archive VM by retiring it
+        testing_vm.mgmt.delete()
+        testing_vm.wait_for_vm_state_change(desired_state='archived', timeout=720,
+                                            from_details=False, from_any_provider=True)
+    else:
+        # Orphan VM by removing provider from CFME
+        provider.delete_if_exists(cancel=False)
+        testing_vm.wait_for_vm_state_change(desired_state='orphaned', timeout=720,
+                                            from_details=False, from_any_provider=True)
+    yield request.param
+
+
+def test_power_options_on_archived_orphaned_vms_all_page(archive_orphan_vm, testing_vm):
+    """This test case is to check Power option drop-down button is disabled on archived and orphaned
+    VMs all page. Also it performs the power operations on vm and checked expected flash messages.
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/2h
+        caseimportance: low
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.9
+        casecomponent: Control
+        tags: power
+        testSteps:
+            1. Add provider infrastructure provider
+            2. Navigate to Archived or orphaned VMs all page
+            3. Select any VM and click on power option drop-down
+        expectedResults:
+            1.
+            2.
+            3. As we don't perform any power operations on archived or orphaned VMs. So power
+               button drop-down should not present there at all or it must show desired flash
+               messages for those actions.
+
+    Bugzilla:
+        1655477, 1686015
+    """
+    infra_vms = testing_vm.appliance.collections.infra_vms
+    if archive_orphan_vm == "archived_vm":
+        view = navigate_to(infra_vms, 'ArchivedAll')
+
+        # Checking power drop down is disabled on 'ArchivedAll' view
+        assert not view.toolbar.power.is_enabled
+
+        # Selecting particular archived vm
+        testing_vm.find_quadicon(from_archived_all=True).check()
+    else:
+        view = navigate_to(infra_vms, 'OrphanedAll')
+
+        # Checking power drop down is disabled on 'OrphanedAll' view
+        assert not view.toolbar.power.is_enabled
+
+        # Selecting particular orphaned vm
+        testing_vm.find_quadicon(from_orphaned_all=True).check()
+
+    # After selecting particular archived/orphaned vm; 'Power' drop down gets enabled.
+    # Reading all the options available in 'power' drop down
+    for action in view.toolbar.power.items:
+        # Performing power actions on archived/orphaned vm
+        view.toolbar.power.item_select(action, handle_alert=True)
+        if action == 'Power On':
+            action = 'Start'
+        elif action == 'Power Off':
+            action = 'Stop'
+        view.flash.assert_message('{} action does not apply to selected items'.format(action))
```

Comment:
Consider clearing the flash messages between toolbar clicks.  There's a method for it:
https://github.com/RedHatQE/widgetastic.patternfly/blob/master/src/widgetastic_patternfly/__init__.py#L245